### ISSUE TITLE
[bug] django.test overridesettings and TestCase.settings are replacing Dynaconf with UserSettingsHolder

### ISSUE THREAD

**Issue by rochacbruno** [rochacbruno/dynaconf#643]
This causes `settings._wrapped._fresh` to be not found as it is a dynaconf attribute

this line on django

https://github.com/django/django/blob/bf5abf1bdcedb15e949db419c61eeec7c88414ea/django/test/utils.py#L470

```py
        override = UserSettingsHolder(settings._wrapped)
        for key, new_value in self.options.items():
            setattr(override, key, new_value)
        self.wrapped = settings._wrapped
        settings._wrapped = override
```

will make the `settings._wrapped` to be the `django.conf.UserSettingsHolder` (not a dynaconf object)

## solution

in this line

https://github.com/rochacbruno/dynaconf/blob/e0225e6f37f7daf3d5015f9d3d9cad7b8ed63681/dynaconf/contrib/django_dynaconf_v2.py#L101

add `if name == 'UserSettingsHolder` and then return a `dynaconf` aware object.

**Comment by rochacbruno** [comment]
Backport this fix to `3.1.z`

**Comment by rochacbruno** [comment]
dup #596
